hyperscan/__init__.py

```diff
@@ -46,14 +46,16 @@ def __init__(self, patterns=None, flags=None, mode=None):
 
     if mode is None:
       mode = self._hs.HS_MODE_BLOCK
-
+    if ids is None:
+      ids = [0] * len(patterns)
```

Comment:
What do you think about making the default a NULL pointer here instead of replicating Hyperscan's default? Seems more robust to me even if you'd have to special case it in _CompilePatterns then.